deps/wrapper.cc

```diff
@@ -182,33 +182,29 @@ ray::core::CoreWorker &_GetCoreWorker() {
     return CoreWorkerProcess::GetCoreWorker();
 }
 
-// https://github.com/ray-project/ray/blob/ray-2.5.1/src/ray/core_worker/test/core_worker_test.cc#L224-L237
-ObjectID put(std::shared_ptr<Buffer> buffer) {
+// Example of using `CoreWorker::Put`: https://github.com/ray-project/ray/blob/ray-2.5.1/src/ray/core_worker/test/core_worker_test.cc#L224-L237
+ObjectID put(const std::shared_ptr<RayObject> object,
+             const std::vector<ObjectID> &contained_object_ids) {
+
     auto &worker = CoreWorkerProcess::GetCoreWorker();
 
-    // Store our string in the object store
+    // Store our data in the object store
     ObjectID object_id;
-    RayObject ray_obj = RayObject(buffer, nullptr, std::vector<rpc::ObjectReference>());
-    RAY_CHECK_OK(worker.Put(ray_obj, {}, &object_id));
-
+    RAY_CHECK_OK(worker.Put(*object, contained_object_ids, &object_id));
     return object_id;
 }
 
-// https://github.com/ray-project/ray/blob/ray-2.5.1/src/ray/core_worker/test/core_worker_test.cc#L210-L220
-std::shared_ptr<Buffer> get(ObjectID object_id) {
+// Example of using `CoreWorker::Get`: https://github.com/ray-project/ray/blob/ray-2.5.1/src/ray/core_worker/test/core_worker_test.cc#L210-L220
+std::shared_ptr<RayObject> get(const ObjectID object_id) {
     auto &worker = CoreWorkerProcess::GetCoreWorker();
 
     // Retrieve our data from the object store
-    std::vector<std::shared_ptr<RayObject>> results;
+    std::vector<std::shared_ptr<RayObject>> objects;
     std::vector<ObjectID> get_obj_ids = {object_id};
-    RAY_CHECK_OK(worker.Get(get_obj_ids, -1, &results));
+    RAY_CHECK_OK(worker.Get(get_obj_ids, -1, &objects));
```

Comment:
kind of orthogonal to this PR but we shouldn't be using `RAY_CHECK*` because when it fails, it exits the process.  better to check + throw.  but we can handle that in a cleanup later.

Reply:
I can make a follow up issue for this. The `CoreWorker::Get` usually returns the `Status` and lets the caller decide what to do. We should probably do the same but as you said that can be done as a follow up.

Reply:
https://github.com/beacon-biosignals/ray_core_worker_julia_jll.jl/issues/84

Reply:
yeah, this is an issue for #78 as well.  I think potentially we could return a tuple or a wrapper of some kind.